scripts/install_server_from_src.sh

```diff
@@ -15,67 +14,133 @@ set -e
 : ${DOWNLOAD_CLIENT_TEMPLATES:=true}
 : ${PYTHON_INSTALL:=true}
 : ${DOCKER:=false}
+: ${INSTALL_BIN:="install"}
+: ${INSTALL_OPTS:="-p -m644"}
+: ${SRC_DIR:="."}
+
+function header()
+{
+  echo ""
+  echo "##########################################################################################"
+  echo "     ${*}";
+  echo "##########################################################################################"
+}
+
+OPTIND=1
+while getopts "h?ls:" opt; do
+    case "$opt" in
+    h|\?)
+        echo "Usage: ./install_server_from_src.sh [OPTIONS]"
+        echo " -s Path to GRR source"
+        echo " -l Install using symlinks to GRR source for installed files"
+        exit 0
+        ;;
+    s)  SRC_DIR=$OPTARG;
+        ;;
+    l)  INSTALL_BIN="ln";
+        INSTALL_OPTS="-snf";
+        ;;
+    esac
+done
+
+#shift $((OPTIND-1))
+#[ "$1" = "--" ] && shift
 
+INSTALL_CMD="$INSTALL_BIN $INSTALL_OPTS"
 if [ "$EUID" -ne 0 ]
   then echo "Please run as root"
   exit 1
 fi
 
+cd $SRC_DIR
 CURDIR=$(basename $PWD)
 if [[ "$CURDIR" != "grr" ]]; then
-  echo "Please run from the grr source directory"
+  echo "Please run from the grr source directory or provide a valid path to the source directory with -s"
   exit 2
 fi
 
+header "Removing Any Old GRR Installs and Install GRR From Source"
 if $PYTHON_INSTALL; then
+  header "Installing GRR Dependencies"
+  # Install GRR dependencies
+  $SRC_DIR/scripts/install_script_ubuntu.sh -dy -r $SRC_DIR/requirements.txt
```

Comment:
Make this its own section (PYTHON_INSTALL was supposed to just be the setup.py install stuff) and turn it off for docker.